packages/orm/src/rules.ts

```diff
@@ -11,7 +11,13 @@ import { MaybePromise, groupBy, maybePromiseThen } from "./utils";
  *
  * Consumers can extend `GenericError` to add fields relevant for their application.
  */
-export type ValidationRuleResult<E extends GenericError> = string | string[] | E | E[] | undefined;
+export type ValidationRuleResult<E extends GenericError> =
+  | string
+  | string[]
+  | E
+  | E[]
+  | { field: string; msg: string }
```

Comment:
@brudil can we make this `{ fieldName: string; message: string }`?

Reply:
...actually chatting with @zgavin , we're thinking `{ field: string, message: string }`, i.e. field as you have, but message spelled out; wdyt? 

Zach had also mentioned maybe typing this as `field: FieldOf<T>` but then we'd have to pass the `T` generic through, which is doable, but :shrug: I'm more ambivalent about that for this PR.